Source/WebCore/contentextensions/ContentExtensionParser.cpp

```diff
@@ -297,19 +297,24 @@ static std::optional<Expected<ContentExtensionRule, std::error_code>> loadRule(c
     if (!action->has_value())
         return makeUnexpected(action->error());
 
-#if ENABLE(DNR_ON_RULE_MATCHED_DEBUG)
+    return { { { WTFMove(trigger.value()), WTFMove(action->value()) } } };
+}
+
+static std::optional<Expected<ContentExtensionRule, std::error_code>> loadRuleIdentifier(const JSON::Object& ruleObject)
+{
     auto identifierValue = ruleObject.getValue("_identifier"_s);
-    if (!identifierValue)
-        return { { { WTFMove(trigger.value()), WTFMove(action->value()), 0 } } };
+    if (!identifierValue || !identifierValue->asValue())
+        return std::nullopt;
 
-    auto identifierInteger = identifierValue->asValue()->asInteger();
-    if (!identifierInteger.has_value())
+    auto identifier = identifierValue->asValue()->asInteger();
+    if (!identifier.has_value())
         return makeUnexpected(ContentExtensionError::JSONInvalidRuleIdentifier);
 
-    return { { { WTFMove(trigger.value()), WTFMove(action->value()), static_cast<uint32_t>(identifierInteger.value()) } } };
-#else
-    return { { { WTFMove(trigger.value()), WTFMove(action->value()) } } };
-#endif
+    auto trigger = loadTrigger(ruleObject);
+    if (!trigger.has_value())
+        return makeUnexpected(trigger.error());
+
+    return { { { WTFMove(trigger.value()), Action { ReportIdentifierAction { static_cast<uint32_t>(identifier.value()) } } } } };
```

Comment:
If we went with actually serializing the identifier from the JSON we would either have to serialize more than 32 bits, limit the max identifier to UINT_MAX, or truncate and lose some values.  Why don't we just serialize the index of the rule in the original JSON array instead and go get the identifier and other action properties later using that index?